RdlMigration/ConvertRDL.cs

```diff
@@ -560,6 +577,50 @@ private XElement CreateDataSourceNode(XDocument doc, DataSourceDefinition curren
             return currNode;
         }
 
+        private void DiscoverSubreports(XDocument doc)
+        {
+            var subreports = doc.Descendants(doc.Root.Name.Namespace + "Subreport");
+            var subreportPaths = new List<string>();
+            foreach (XElement subreport in subreports)
+            {
+                var subreportName = subreport.Elements().First().Value;
+                var subreportPath = Path.Combine(rootFolder, subreportName);
+                subreportPath = Path.GetFullPath(subreportPath).Replace("\\", "/");
+                subreportPath = subreportPath.Substring(subreportPath.IndexOf('/'));
+                subreportName = Path.GetFileName(subreportPath);
+                
+                try
+                {
+                    if (rdlFileIO.IsFolder(subreportPath))
+                    {
+                        Trace($"SUBREPORT FAIL : {subreportPath} Subreport does not exist");
+                        continue;
+                    }
+                }
+                catch (Exception)
+                {
+                    Trace($"SUBREPORT FAIL : {subreportPath} Subreport does not exist");
+                    continue;
+                }
+
+                if (!reportNameMap.TryGetValue(subreportName, out string existingSubreportPath))
```

Comment:
what is the purpose of this out string?

Reply:
if not using out, then why not use ContainsKey method?

Reply:
you should log this in the ELSE block

Reply:
Sorry, the out value was used in a previous case that's now removed. I'll change it to ContainsKey.